src/features/basket/basketSlice.js

```diff
@@ -1,12 +1,82 @@
 import { createSlice } from "@reduxjs/toolkit";
+import * as image from "images";
 
 export const basketSlice = createSlice({
   name: "basket",
-  initialState: {},
-  reducers: {},
+  initialState: {
+    status: "iddle",
+    items: [
+      {
+        picture: image.headphones,
+        name: "Headphones",
+        price: 11.9,
+        id: 0,
+        value: 2,
+      },
+      {
+        picture: image.mobilePhone,
+        name: "Mobile phone",
+        price: 20.99,
+        id: 1,
+        value: 1,
+      },
+    ],
+    shipping: 23.8,
+    subtotal: 0,
+    grandTotal: 0,
+  },
+  reducers: {
+    deleteItem: (state, action) => {
+      state.items = [...state.items].filter(
+        (item) => item.id !== action.payload
+      );
+    },
+    setSubtotal: (state, action) => {
+      state.subtotal = action.payload;
+    },
+    setGrandTotal: (state, action) => {
+      state.grandTotal = action.payload;
+    },
+    setShipping: (state, action) => {
+      state.shipping = action.payload;
+    },
+    setValue: (state, action) => {
+      if (action.payload.value < 1) {
+        return;
+      }
+
+      state.items = [...state.items].map((item) => ({
+        ...item,
+        value:
+          item.id === action.payload.id
+            ? parseInt(action.payload.value)
+            : item.value,
+      }));
+    },
+    incrementValue: (state, action) => {
```

Comment:
zamiast incrementValue i decrementValue mozesz zrobić jedna metode increaseValue i przekazywać id i value (number, w tym przypadku przekazalbys +1, a tam -1). Finalnie to samo, wiadomo o co chodzi, mniej kodu, bardziej elastyczne na przyszlosc

Reply:
Wiem, nie wiem po kiego wała zrobiłem tak. wtedy bym też zwracał obiekt {id: x, value, y} i destrukturyzację bym zastosował tak jak powyżej sugierowałeś.

Reply:
Skipped.